pkg/api/customization/cluster/validator.go

```diff
@@ -37,35 +38,49 @@ type Validator struct {
 }
 
 func (v *Validator) Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
-	var spec v3.ClusterSpec
+	var clusterSpec v3.ClusterSpec
+	var clientClusterSpec mgmtclient.Cluster
+	logrus.Tracef("Validator: data: %+v\n", data)
+	if err := convert.ToObj(data, &clusterSpec); err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, "Cluster spec conversion error")
+	}
 
-	if err := convert.ToObj(data, &spec); err != nil {
+	if err := convert.ToObj(data, &clientClusterSpec); err != nil {
 		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, "Cluster spec conversion error")
 	}
 
+	logrus.Tracef("Validator: clusterSpec: %+v\n", &clusterSpec)
```

Comment:
We can't log this, it can/will have creds in it. Logging entire objects even in trace is not safe.